docs/docsite/rst/reference_appendices/general_precedence.rst

```diff
@@ -0,0 +1,136 @@
+.. contents:: Topics
+
+
+How Ansible resolves configuration, data and keywords
+=====================================================
+
+Ansible has many ways to specify certain information, specially when dealing with 'connections'
+as this information can be very sensitive to different factors and users have many different contexts to
+deal with.
+
+This desgin allows Ansible to deal with heterogeneous environments, i.e when a host or group of hosts will have different 'connection' information than the others and should always be accessed in this way.
+You are not mandated to use the variables, but most people do out of convenience and then can get confused by their 'high precedence' and how to override them.
+
+In general the precedence is simple, when things are defined at the same level, 'last wins', as per the levels
+this is the most general list, from least to most:
+
+ Configuration
+ Command Line options
+ Playbooks
+ Connection variables
+
+Some of these entries are not self evident and others also have their own internal levels and precedence, see them expanded below.
+
+
+.. _what_is_configuration:
+
+Configuration
+`````````````
+
+This mainly reffers to 2 things, 'the ansible configuration file' and environment variables. The internal precedence is as follows (from least to most):
+
+   configuration file (only first found used, they don't merge)
+      "value of ANSIBLE_CONFIG"
+      `./ansible.cfg`
+      `~/.ansible.cfg`
+      `/etc/ansible/ansible.cfg`
+
+   environment variable (this just follows normal shell precedence, last time defined overwrites previous values)
+
+You can use the ``ansible-config`` command line utility to see the current value of a configuration item and where it came from.
+
+
+.. _command_line_options_precedence:
+
+Command line options
+`````````````````````
+
+The internal precedence is simple, but depends on the parameter used, 'last defined' wins (connects as carol, not mike)::
+
+      ansible -u mike -m ping myhost -u carol
+
+Unless the parameter allows for multiple entries and then they get appended (hosts are from inventory1 + inventory2)::
+
+   ansible -i /path/inventory1 -i /path/inventory2 -m ping all
+
+The help for each tool specifies which parameters allow for multiple entries.
+
+Most of these options are supposed to deal with generic settings, but over time some very connection and strategy specific ones were added, they are also the main source of confusion as most of them have low precedence,
+they are mostly meant to override configuration (which does NOT include playboks, inventory nor variables).
+
+The one exception is 'extra vars' which sets variables with the highest precedence .. and since 'conneciton variables' are also the highest precedence it creates a short circut to it's own options::
```

Comment:
```suggestion
The one exception is 'extra vars' which sets variables with the highest precedence .. and since 'conneciton variables' are also the highest precedence it creates a conflict with its own options::
```